packages/api/src/decorator/private/createDecoratorComposer.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { type ReactNode, useMemo } from 'react';
+import {
+  ActivityBorderDecoratorMiddlewareProvider,
+  activityBorderDecoratorTypeName,
+  type ActivityBorderDecoratorMiddleware
+} from './ActivityBorderDecoratorMiddleware';
+
+type DecoratorMiddlewareByInit = {
+  [activityBorderDecoratorTypeName]: ActivityBorderDecoratorMiddleware;
+};
+
+type DecoratorMiddlewareInit = keyof DecoratorMiddlewareByInit;
+
+export type DecoratorComposerComponent = (
+  props: Readonly<{
+    children: ReactNode;
+    middleware: DecoratorMiddleware[];
+  }>
+) => React.JSX.Element;
+
+export type DecoratorMiddleware = (
+  init: DecoratorMiddlewareInit
+) => ReturnType<ActivityBorderDecoratorMiddleware> | false;
+
+const initMiddlewares = <Init extends DecoratorMiddlewareInit>(
```

Comment:
Could be a cool utility. 🤣

Reply:
Should it part of the `templateMiddleware`... unsure if yes or no...

We can refactor the into utility in next PR too.